shared/chat/blocking/block-modal/index.tsx

```diff
@@ -0,0 +1,283 @@
+import * as React from 'react'
+import * as Kb from '../../../common-adapters'
+import * as Styles from '../../../styles'
+
+export type BlockType = 'chatBlocked' | 'followBlocked'
+type BlocksForUser = {chatBlocked?: boolean; followBlocked?: boolean}
+
+export type NewBlocksMap = Map<string, BlocksForUser>
+type State = {
+  blockTeam: boolean
+  extraNotes: string
+  finishClicked: boolean
+  includeTranscript: boolean
+  newBlocks: NewBlocksMap
+  reportReason: string
+  shouldReport: boolean
+}
+
+export type Props = {
+  adderUsername: string
+  blockByDefault?: boolean
+  convID?: string
+  isBlocked: (username: string, which: BlockType) => boolean
+  onCancel: () => void
+  onFinish: (newBlocks: NewBlocksMap, blockTeam: boolean) => void
+  otherUsernames?: Array<string>
+  refreshBlocks: () => void
+  teamname?: string
+}
+
+type CheckboxRowProps = {
+  checked: boolean
+  info?: string
+  more?: React.ReactNode
+  onCheck: (boolean) => void
+  text: React.ReactNode
+}
+const CheckboxRow = (props: CheckboxRowProps) => {
+  const [infoShowing, setInfoShowing] = React.useState(false)
+  return (
+    <>
+      <Kb.Box2 direction="horizontal" alignItems="center" fullWidth={true} style={styles.checkBoxRow}>
+        <Kb.Switch
+          color="red"
+          gapSize={Styles.globalMargins.tiny}
+          label={props.text}
+          labelSubtitle={infoShowing ? props.info : undefined}
+          on={props.checked}
+          onClick={() => props.onCheck(!props.checked)}
+          style={styles.shrink}
+        />
+        <Kb.Box style={styles.iconBox} />
+        {props.info && !infoShowing && (
+          <Kb.Icon type="iconfont-question-mark" color="grey" onClick={() => setInfoShowing(true)} />
+        )}
+      </Kb.Box2>
+    </>
+  )
+}
+
+type ReportOptionsProps = {
+  extraNotes: string
+  includeTranscript: boolean
+  reason: string
+  setExtraNotes: (text: string) => void
+  setIncludeTranscript: (checked: boolean) => void
+  setReason: (reason: string) => void
+  showIncludeTranscript: boolean
+}
+const reasons = ["I don't know this person", 'Spam', 'Harassment', 'Obscene material', 'Other...']
+const ReportOptions = (props: ReportOptionsProps) => {
+  return (
+    <>
+      {reasons.map(reason => (
+        <Kb.RadioButton
+          key={reason}
+          label={reason}
+          onSelect={() => props.setReason(reason)}
+          selected={props.reason === reason}
+          style={styles.radioButton}
+        />
+      ))}
+      <Kb.Box style={styles.feedback}>
```

Comment:
This could use some padding below
<img width="465" alt="image" src="https://user-images.githubusercontent.com/11968340/68778766-9ef7df00-0601-11ea-8917-245d534d82a4.png">